### ISSUE TITLE
Error on download tab delimeted

### ISSUE THREAD

**Issue by keyurshah** [jlevers/selling-partner-api#21]
Thank you for this great library!

I am uploading a price and quantity feed upload. The upload content-type is `text-tab/separated-values`

 That works great, and now I'm trying to download the results using the recommended code.

```
        //upload feed
        $feed_id = 60619018758;

        $api  = new FeedsApi();
        $feed = $api->getFeed($feed_id);

        $feed_document_id = $feed->getPayload()->getResultFeedDocumentId();

        $feed_document = $api->getFeedDocument($feed_document_id);
        $docToDownload       = new Document($feed_document->getPayload(), "text/tab-separated-values");
        $contents = $docToDownload->download();
```

everything seems fine until the download step, which gives me the error:

```
   ErrorException 

  Undefined offset: 1

 at C:\myfiles\webserver\www\myproject\vendor\jlevers\selling-partner-api\lib\Document.php:92
     88▕             $header = fgetcsv($bareStream, 0, $sep);
     89▕             while (($line = fgetcsv($bareStream, 0, $sep)) !== false) {
     90▕                 $row = [];
     91▕                 foreach ($line as $idx => $val) {
  ➜  92▕                     $row[$header[$idx]] = $val;
     93▕                 }
     94▕                 $data[] = $row;
     95▕             }
     96▕             $this->data = $data;
```

If i change the line to `text/csv` then I don't get an error. But the data still seems to be in tab format when doing a dump
 
```
        $docToDownload       = new Document($feed_document->getPayload(), "text/csv");
```

outputs on dump:

```
"""
Feed Processing Summary:\n
\tNumber of records processed\t\t31\n
\tNumber of records successful\t\t31\n
\n
"""
```

so there seems to be an issue in parsing tab delimited results possibly?

thanks!

**Comment by keyurshah** [comment]
Thanks! That would be appreciated.  Too bad the response isn’t more consistent for parsing.

**Comment by cpiggott** [comment]
Would it be worth adding these types:

`"text/tab-separated-values"` and `"text/csv"` as constants to the library as well to more easily use them.

**Comment by jlevers** [comment]
Good idea -- I'm adding XLSX support sometime soon for the reports of that type. Ideally, there would just be a file of report type array constants where the user could just specify a constant and not have to worry about the content type.

Something like:

```php
const CT_TAB = 'text/tab-separated-values'
// ...
const GET_FLAT_FILE_OPEN_LISTINGS_DATA = [
    'report' => 'GET_FLAT_FILE_OPEN_LISTINGS_DATA',
    'contentType' => CT_TAB
];
// ...
```

**Comment by jlevers** [comment]
Update: the report and feed type constants were added in v3.0.0.

Re: the original issue (parsing feed result documents correctly), once the Amazon issue I'm running into (see #50) is dealt with, I'll update the parser to handle the extra data at the beginning of feed result documents.

**Comment by jlevers** [comment]
This should be resolved by `v3.0.2`.
